internal/datatypes/core_test.go

```diff
@@ -0,0 +1,65 @@
+package datatypes
+
+import (
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var datatypeTests = []struct {
```

Comment:
oh nice! I didn't expect you'd be able to get them all as data. But yes, much clearer.